public/app/features/integrated-alerting/components/AlertRules/AddAlertRuleModal/AddAlertRuleModal.tsx

```diff
@@ -0,0 +1,159 @@
+      setChannelsOptions(formatChannelsOptions(channelsListResponse));
+      const templatesListResponse = await AlertRuleTemplateService.list();
+      setTemplateOptions(formatTemplateOptions(templatesListResponse.templates));
+    } catch (e) {
+      logger.error(e);
+    }
+  };
+
+  useEffect(() => {
+    getData();
+  }, []);
+
+  const onSubmit = async (values: AddAlertRuleFormValues) => {
+    try {
+      await AlertRulesService.create(formatCreateAPIPayload(values));
+      setVisible(false);
+      appEvents.emit(AppEvents.alertSuccess, [Messages.addSuccess]);
+      getAlertRules();
+    } catch (e) {
+      logger.error(e);
+    }
+  };
+
+  return (
+    <Modal
+      title={Messages.title}
+      isVisible={isVisible}
+      onClose={() => setVisible(false)}
+      data-qa="add-alert-rule-modal"
+    >
+      <Form
+        onSubmit={onSubmit}
+        render={({ handleSubmit, valid, pristine, submitting }) => (
+          <form className={styles.form} onSubmit={handleSubmit} data-qa="add-alert-rule-modal-form">
+            {/* TODO: polish this up */}
+            <Field name="template" validate={required}>
+              {({ input }) => (
+                <>
+                  <label className={styles.label} data-qa="type-field-label">
+                    {Messages.templateField}
+                  </label>
+                  <Select className={styles.select} options={templateOptions} {...input} data-qa="template-select" />
+                </>
+              )}
+            </Field>
+
+            <TextInputField label={Messages.nameField} name="name" validators={[required]} />
+
+            <TextInputField label={Messages.thresholdField} name="threshold" validators={[required]} />
+
+            <NumberInputField label={Messages.durationField} name="duration" validators={[required]} />
+
+            <Field name="severity" validate={required}>
+              {({ input }) => (
+                <>
+                  <label className={styles.label} data-qa="type-field-label">
+                    {Messages.severityField}
+                  </label>
+                  <Select
+                    className={styles.select}
+                    options={SEVERITY_OPTIONS}
+                    {...input}
+                    data-qa="severity-multiselect"
+                  />
+                </>
+              )}
+            </Field>
+
+            <TextareaInputField label={Messages.filtersField} name="filters" validators={[required]} />
+
+            <Field name="notificationChannels">
+              {({ input }) => (
+                <>
+                  <label className={styles.label} data-qa="type-field-label">
+                    {Messages.channelField}
+                  </label>
+                  <MultiSelect
+                    className={styles.select}
+                    options={channelsOptions}
+                    {...input}
+                    data-qa="notification-channels-multiselect"
```

Comment:
```suggestion
                    data-qa="notificationChannels-multiselect"
```
I think for automation purposes the best would be to use the name. This is the way core components are coded.

Reply:
Done